pkg/dashboards/dashboards.go

```diff
@@ -432,6 +387,10 @@ func updateDashboard(id string, dName string, dashboardDetails map[string]interf
 		return errors.New("updateDashboard: Error fetching dashboard details")
 	}
 
+	if currentCreatedAt, exists := currentDashboardDetails["createdAt"]; exists {
+		dashboardDetails["createdAt"] = currentCreatedAt
+	}
```

Comment:
nit: It would probably be a little better to make `currentDashboardDetails` an instance of a struct type and then have `createdAt` be a field. That way, if there's a typo in "createdAt" we'd get a compile-time error instead of it silently failing

Reply:
If we use a struct, won't it require updating every time the UI adds a new field?

Reply:
Yes. If you think that cost outweighs the benefit of detecting typos easier, then you can keep what you have. You might have a better judgement on this than me since you're more familiar with how the frontend works and how often new fields would get added.

If you do change it to a struct, I think there's some way to make the json unmarshalling strict so it will error if there's a field it doesn't expect, which might be useful. But I'll approve this and you can merge it if you want